PMSCore/StateManager.cs

```diff
@@ -283,6 +283,70 @@ public EntityState UpdateEntityProperty(string entityId, EntityProperty property
         return EntityState.Forbidden;
     }
 
+    /// <summary>
+    /// Updates a user property (userId | Permission)
+    /// </summary>
+    /// <param name="userId"></param>
+    /// <param name="property"></param>
+    /// <param name="value"></param>
+    /// <returns></returns>
+    public EntityState UpdateUserProperty(string userId, UserProperty property, object value)
+    {
+        if (!_users.ContainsKey(userId))
```

Comment:
changing if (!_users.ContainsKey(userId)) to if (!_users.TryGetValue(userId, out var user)) to check for the user’s existence and return the user object at the same time

Reply:
Nice one changed accordingly